firefox-ios/Client/Frontend/TrackingProtection/CertificatesViewController.swift

```diff
@@ -0,0 +1,280 @@
+            }
+            headerStackView.addArrangedSubview(button)
+        }
+
+        view.addSubview(headerStackView)
+
+        NSLayoutConstraint.activate([
+            headerStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor,
+                                                     constant: CertificatesUX.headerStackViewMargin),
+            headerStackView.trailingAnchor.constraint(equalTo: view.trailingAnchor,
+                                                      constant: -CertificatesUX.headerStackViewMargin),
+            headerStackView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor,
+                                                 constant: CertificatesUX.headerStackViewTopMargin)
+        ])
+    }
+
+    private func setupCertificatesTableView() {
+        certificatesTableView.delegate = self
+        certificatesTableView.dataSource = self
+        view.addSubview(tableViewTopSpacer)
+        view.addSubview(certificatesTableView)
+        NSLayoutConstraint.activate([
+            tableViewTopSpacer.topAnchor.constraint(equalTo: headerStackView.bottomAnchor,
+                                                    constant: CertificatesUX.tableViewSpacerTopMargin),
+            tableViewTopSpacer.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableViewTopSpacer.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            tableViewTopSpacer.heightAnchor.constraint(equalToConstant: CertificatesUX.tableViewSpacerHeight),
+
+            certificatesTableView.topAnchor.constraint(equalTo: tableViewTopSpacer.bottomAnchor,
+                                                       constant: CertificatesUX.tableViewTopMargin),
+            certificatesTableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            certificatesTableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            certificatesTableView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+    }
+
+    @objc
+    private func certificateButtonTapped(_ sender: UIButton) {
+        viewModel.selectedCertificateIndex = sender.tag
+        certificatesTableView.reloadData()
+    }
+
+    // MARK: - UITableViewDataSource
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard !viewModel.certificates.isEmpty else { return 0 }
+        return 4
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: CertificatesCell.cellIdentifier, for: indexPath)
+                as? CertificatesCell else {
+            return UITableViewCell()
+        }
+
+        guard !viewModel.certificates.isEmpty else {
+            return cell
+        }
+
+        let certificate = viewModel.certificates[viewModel.selectedCertificateIndex]
+
+        switch indexPath.row {
```

Comment:
This is super close!

One very small nitpick to make things look cleaner:

```
switch CertificatesItemType(rawValue: indexPath.row) {
        case .subjectName:
        ...
        case .issuerName:
        ...
        ...
        ...
        case .none: // Deals with the nil case. 
            break
}
```
I also suggested `.none` for the nil case. `default` is ok but the compiler will not give you a warning if, in the future, someone adds a new `CertificatesItemType` case that isn't handled here.

FYI if you're curious, two examples of this pattern are in `LegacyInactiveTabCell` line 146 and `AddCredentialViewController` line 156.